monai/networks/blocks/mednext_block.py

```diff
@@ -0,0 +1,233 @@
+# You may obtain a copy of the License at
+#     http://www.apache.org/licenses/LICENSE-2.0
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# Portions of this code are derived from the original repository at:
+# https://github.com/MIC-DKFZ/MedNeXt
+# and are used under the terms of the Apache License, Version 2.0.
+
+from __future__ import annotations
+
+import torch
+import torch.nn as nn
+import torch.nn.functional as F
+
+
+class MedNeXtBlock(nn.Module):
+
+    def __init__(
+        self,
+        in_channels: int,
+        out_channels: int,
+        exp_r: int = 4,
+        kernel_size: int = 7,
+        do_res: int = True,
+        norm_type: str = "group",
+        n_groups: int or None = None,
+        dim="3d",
+        grn=False,
+    ):
+
+        super().__init__()
+
+        self.do_res = do_res
+
+        assert dim in ["2d", "3d"]
+        self.dim = dim
+        if self.dim == "2d":
+            conv = nn.Conv2d
+        else:
+            conv = nn.Conv3d
+
+        # First convolution layer with DepthWise Convolutions
+        self.conv1 = conv(
+            in_channels=in_channels,
+            out_channels=in_channels,
+            kernel_size=kernel_size,
+            stride=1,
+            padding=kernel_size // 2,
+            groups=in_channels if n_groups is None else n_groups,
+        )
+
+        # Normalization Layer. GroupNorm is used by default.
+        if norm_type == "group":
+            self.norm = nn.GroupNorm(num_groups=in_channels, num_channels=in_channels)
+        elif norm_type == "layer":
+            self.norm = LayerNorm(normalized_shape=in_channels, data_format="channels_first")
+
+        # Second convolution (Expansion) layer with Conv3D 1x1x1
+        self.conv2 = conv(in_channels=in_channels, out_channels=exp_r * in_channels, kernel_size=1, stride=1, padding=0)
+
+        # GeLU activations
+        self.act = nn.GELU()
+
+        # Third convolution (Compression) layer with Conv3D 1x1x1
+        self.conv3 = conv(
+            in_channels=exp_r * in_channels, out_channels=out_channels, kernel_size=1, stride=1, padding=0
+        )
+
+        self.grn = grn
+        if self.grn:
+            if dim == "2d":
+                self.grn_beta = nn.Parameter(torch.zeros(1, exp_r * in_channels, 1, 1), requires_grad=True)
+                self.grn_gamma = nn.Parameter(torch.zeros(1, exp_r * in_channels, 1, 1), requires_grad=True)
+            else:
+                self.grn_beta = nn.Parameter(torch.zeros(1, exp_r * in_channels, 1, 1, 1), requires_grad=True)
+                self.grn_gamma = nn.Parameter(torch.zeros(1, exp_r * in_channels, 1, 1, 1), requires_grad=True)
+
```

Comment:
I think this is easier to understand, but the current one is fine as well.
```suggestion
            if dim == "2d":
                grn_parameter_shape = (1,1)
            elif dim == "3d":
                grn_parameter_shape = (1,1,1)
            else:
                raise ValueError()

            grn_parameter_shape = (1, exp_r * in_channels,) + grn_parameter_shape
            self.grn_beta = nn.Parameter(torch.zeros(grn_parameter_shape), requires_grad=True)
            self.grn_gamma = nn.Parameter(torch.zeros(grn_parameter_shape), requires_grad=True)

        
```